ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/ActionManager.java

```diff
@@ -93,5 +50,5 @@ private ActionManager()
 
 	private static final ActionManager _theManager = new ActionManager();
 
-	private static final Map<Uid, Lifetime> _allActions = new ConcurrentHashMap<Uid, Lifetime>();
+	private static final Map<Uid, BasicAction> _allActions = new ConcurrentHashMap<>();
```

Comment:
Unfortunately it's in a public class so you'd need to `@Deprecate` first.

Reply:
Yeah, I also thought ActionManager was a bit of an overkill and your proposal to deprecate it sounds good to me. I will go ahead, and deprecate it (and I will add your suggestion to the JBTM). Thanks Michael

Reply:
Well...while I was deprecating `ActionManager`, I recalled what I thought before lunch: this class could be useful to highlight the distinction between who keeps track of Actions and the Actions themselves, even though, in Arjuna, the foundation Action is `BasicAction` (and that won't change).

What I'm trying to say is that the relation between `ActionManager` and `BasicAction` can be thought (more or less) like the relation between `Collection/List/Map/etc.` and `Collections` in the standard Java library. IOW, some generic operations (min, max, binary search, etc.) are implemented in `Collections` but they are applied on `List/ Map/Collection/etc.`. It is very true that the foundation Action in Arjuna is `BasicAction` but, in future, we might want to make `ActionManager` track different kind of Actions (e.g. `AtomicAction`). WDYT?

Reply:
@mmusgrov WDYT to merge this PR and open a new JBTM to track/discuss the deprecation of `ActionManager`?

Reply:
So I agree, let's not deprecate it, in particular any external users would need an alternative and I don't think we would should encourage them to use BasicAction directly even though BasicAction is available.

Reply:
I'm gonna merge this then. Thanks for your review, @mmusgrov and @marcosgopen

Reply:
But since you have landed on this class I think we should prepare the way by maintaining the Action map in BasicAction for the following reason:

- calls from BasicAction to ActionMap to update the map happen twice on every transaction
- calls to ActionManager are rare (during recovery and during the interogation of the stats

and since we optimise for TPS, which on good hardware and on one of the faster stores can be thousands of calls per second.

@jmfinelli Could you follow up on this PR with that change please?

Reply:
https://issues.redhat.com/browse/JBTM-3901